src/test/test-copy.c

```diff
@@ -436,6 +436,72 @@ TEST_RET(copy_holes) {
         return 0;
 }
 
+TEST_RET(copy_holes_with_gaps) {
+        _cleanup_(unlink_tempfilep) char fn[] = "/var/tmp/test-copy-hole-with-gaps-fd-src-XXXXXX";
+        _cleanup_(unlink_tempfilep) char fn_copy[] = "/var/tmp/test-copy-hole-with-gaps-fd-tgt-XXXXXX";
```

Comment:
Please replace this with something like:

```
_cleanup_(rm_rf_physical_and_freep) char *t = NULL;
_cleanup_close_ int fd = -EBADF, fd_copy = -EBADF;

assert_se((tfd = mkdtemp_open(NULL, 0, &t)) >= 0);
assert_se((fd = openat(tfd, "src", ...) >= 0);
assert_se((fd_copy = openat(tfd, "dst", ...) >= 0);
```

Reply:
I copied these from the test above. If that is no good then I'm happy to apply your suggestions.

Reply:
What I suggested is how we try to do things for new tests. The test you copied from just hasn't been migrated to that pattern yet.

Reply:
Done